modules/hostap/src/supp_api.c

```diff
@@ -611,11 +756,44 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 				goto out;
 			}
 
-			if (!wpa_cli_cmd_v("set_network %d eap TLS",
-					   resp.network_id)) {
+			if (!wpa_cli_cmd_v("set_network %d eap %s", resp.network_id, method)) {
 				goto out;
 			}
 
+			if (params->security == WIFI_SECURITY_TYPE_EAP_PEAP_MSCHAPV2 ||
+			    params->security == WIFI_SECURITY_TYPE_EAP_PEAP_GTC ||
+			    params->security == WIFI_SECURITY_TYPE_EAP_PEAP_TLS) {
+				snprintf(phase1, sizeof(phase1),
```

Comment:
I would recommend using `snprintk` here as that should have smaller memory consumption

Reply:
Updated as comment.